### PR TITLE
[AIP-131] Enforce a consistent method signature.

### PR DESCRIPTION
**Pull request by lukesneeringer**
Uses #431 as a base.

### CODE REVIEW

File: rules/aip0131/method_signature.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+				Message: fmt.Sprintf(
+					"Get methods should include `(google.api.method_signature) = %q`",
+					"name",
+				),
+				Descriptor: m,
+			}}
+		}
+
+		// Check if the signature is wrong.
+		if diff := cmp.Diff(signatures[0], []string{"name"}); diff != "" {
```
Comment (mingzhi) [inline rules/aip0131/method_signature.go]: Should we just write `len(signature[0]) == !1 && signature[0][0] == "name"`? Using `cmp` seems to be heavy here.

Comment (lukesneeringer) [reply rules/aip0131/method_signature.go]: I think `cmp.Diff` is the standard/preferred way to do this sort of thing.

Comment (mingzhi) [reply rules/aip0131/method_signature.go]: I would prefer `cmp` in testing. But using it probably is overkilled here.

Comment (lukesneeringer) [reply rules/aip0131/method_signature.go]: Okay, using `reflect.DeepEqual` instead.

**Pull request merged**